Tests/Functional/Utilities.swift

```diff
@@ -73,11 +77,34 @@ enum Configuration {
     case Release
 }
 
+private var globalSymbolInMainBinary = 0
+
+
+func swiftBuildPath() -> String {
+#if os(OSX)
+    for bundle in NSBundle.allBundles() where bundle.bundlePath.hasSuffix(".xctest") {
+        return Path.join(bundle.bundlePath.parentDirectory, "swift-build")
+    }
+    fatalError()
+#else
+    return Path.join(try! Process.arguments.first!.abspath().parentDirectory, "swift-build")
+#endif
+}
+
+
 func executeSwiftBuild(chdir: String, configuration: Configuration = .Debug, printIfError: Bool = false, Xld: [String] = []) throws -> String {
-    let toolPath = Resources.findExecutable("swift-build")
     var env = [String:String]()
     env["SWIFT_BUILD_TOOL"] = getenv("SWIFT_BUILD_TOOL")
-    var args = [toolPath, "--chdir", chdir]
+#if Xcode
+    if env["SWIFT_EXEC"] == nil {
+        let bindir = Path.join(getenv("XCODE_DEFAULT_TOOLCHAIN_OVERRIDE")!, "usr/bin")
+        env["SWIFT_EXEC"] = Path.join(bindir, "swiftc")
+        env["SWIFT_BUILD_TOOL"] = Path.join(bindir, "swift-build-tool")
+    } else {
+        fatalError("HURRAY! This is fixed")
```

Comment:
It seems my comment got deleted (by me, bad me). There's a bug in Xcode currently where `SWIFT_EXEC` is not set for tests, I have a radar open for it.